Build for OS X

OS X is failing with:

```
Undefined symbols for architecture x86_64:

      "_PyArg_ParseTupleAndKeywords", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyCObject_AsVoidPtr", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_AsVoidPtr in fortranobject.o

      "_PyCObject_FromVoidPtr", referenced from:

          _fortran_getattr in fortranobject.o

          _F2PyCapsule_FromVoidPtr in fortranobject.o

      "_PyCObject_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_Check in fortranobject.o

      "_PyDict_DelItemString", referenced from:

          _fortran_setattr in fortranobject.o

      "_PyDict_GetItemString", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyDict_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyDict_SetItemString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_getattr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

          _PyFortranObject_New in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyErr_Clear", referenced from:

          _fortran_repr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Format", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyErr_NewException", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyErr_NoMemory", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyErr_Occurred", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Print", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_SetString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyExc_AttributeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_setattr in fortranobject.o

      "_PyExc_ImportError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyExc_RuntimeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyExc_TypeError", referenced from:

          _fortran_call in fortranobject.o

          _array_from_pyobj in fortranobject.o

      "_PyExc_ValueError", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyImport_ImportModule", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyMem_Free", referenced from:

          _fortran_dealloc in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyMem_Malloc", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyModule_GetDict", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyOS_snprintf", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyObject_GetAttrString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

      "_PyString_AsString", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_ConcatAndDel", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyString_FromFormat", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_FromString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyString_FromStringAndSize", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyType_IsSubtype", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyType_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_BuildValue", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_FindMethod", referenced from:

          _fortran_getattr in fortranobject.o

      "_Py_InitModule4_64", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "__PyObject_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

      "__Py_NoneStruct", referenced from:

          _fortran_getattr in fortranobject.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_main", referenced from:

          start in crt1.10.6.o

    ld: symbol(s) not found for architecture x86_64

    collect2: error: ld returned 1 exit status

    Undefined symbols for architecture x86_64:

      "_PyArg_ParseTupleAndKeywords", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyCObject_AsVoidPtr", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_AsVoidPtr in fortranobject.o

      "_PyCObject_FromVoidPtr", referenced from:

          _fortran_getattr in fortranobject.o

          _F2PyCapsule_FromVoidPtr in fortranobject.o

      "_PyCObject_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_Check in fortranobject.o

      "_PyDict_DelItemString", referenced from:

          _fortran_setattr in fortranobject.o

      "_PyDict_GetItemString", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyDict_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyDict_SetItemString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_getattr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

          _PyFortranObject_New in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyErr_Clear", referenced from:

          _fortran_repr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Format", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyErr_NewException", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyErr_NoMemory", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyErr_Occurred", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Print", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_SetString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyExc_AttributeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_setattr in fortranobject.o

      "_PyExc_ImportError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyExc_RuntimeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyExc_TypeError", referenced from:

          _fortran_call in fortranobject.o

          _array_from_pyobj in fortranobject.o

      "_PyExc_ValueError", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyImport_ImportModule", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyMem_Free", referenced from:

          _fortran_dealloc in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyMem_Malloc", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyModule_GetDict", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyOS_snprintf", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyObject_GetAttrString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

      "_PyString_AsString", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_ConcatAndDel", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyString_FromFormat", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_FromString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyString_FromStringAndSize", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyType_IsSubtype", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyType_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_BuildValue", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_FindMethod", referenced from:

          _fortran_getattr in fortranobject.o

      "_Py_InitModule4_64", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "__PyObject_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

      "__Py_NoneStruct", referenced from:

          _fortran_getattr in fortranobject.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_main", referenced from:

          start in crt1.10.6.o

    ld: symbol(s) not found for architecture x86_64

    collect2: error: ld returned 1 exit status

    error: Command "/Users/travis/miniconda/envs/_build/bin/gfortran -Wall -g -arch x86_64 build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_tmodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/f90wrap/sizeoffortran.o -L/Users/travis/miniconda/envs/_build/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5 -L/Users/travis/miniconda/envs/_build/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_t.so" failed with exit status 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,44 @@
+# This file was generated automatically from conda-smithy. To update this configuration,
+# update the conda-forge.yaml and/or the recipe/meta.yaml.
+
+language: generic
+
+os: osx
+
+env:
+  matrix:
+    
+    - CONDA_NPY=110  CONDA_PY=27
+    - CONDA_NPY=111  CONDA_PY=27
+    - CONDA_NPY=110  CONDA_PY=34
+    - CONDA_NPY=111  CONDA_PY=34
+    - CONDA_NPY=110  CONDA_PY=35
+    - CONDA_NPY=111  CONDA_PY=35
+  global:
+    # The BINSTAR_TOKEN secure variable. This is defined canonically in conda-forge.yml.
+    - secure: "qdH+cXC0ePyxmqILnJEglAU0YEwunIqzBXHc8tEgCvZGCgybGb4d44QEBsuER8l1VBqRtQh9mjfof+ElWYhsJarmjxXcnAK+LytWrRn7FvdQjrEgHZTcXOua77iTVGNTsK0r91/YbutT+spmRIec/KJOP2gmaDKHYh9gAnwJA6ITqbs851OoRxY7hgEaDCt/9XQutjs/cD5FDkWBfpUqY3++8mOkmIewQ6E4SOoeFdz3G40hBf6r4FtwD47zSM4Xg6Z442ZoGMlFY6fg9SedwB/fXZ8zRv578QyQlZQFtYDD44zflIHWrGhTAztLRtaCQdrcmeB/xIIqbXKcSvF9mPK7AUSUw5AquXIwKA8tOOPoEwz79qoK2YAauITbHtC0Mys/u08cN/ZbuSD3JRCWxF1wjlPpVL7HMPROyGM46gDeA0Q5yR2Y+pWufZSPC2WnjFIf4fPCMmOLJ8/FTFON2Hc8dtINKmdVPgPjSiRuQiszsiGFCFWm7CsNQFAZKYMhNvrbaqrsvVElFvMoPs1Mjmmgj2KD79c29wX4dHj0ARNOTBHzvQDimc0GbW3ruuL4XqakLv9tPRBj6elmGT7LiM2WeXfXFTtUBcl4vIU7CE9kYENL9gVcfxDN19/L9r/nSRsd1JZNYxJp19OU5gjh1jmnt+9z56Cx06+PczwjGGg="
+
+
+before_install:
+    # Remove homebrew.
+    - brew remove --force $(brew list)
+    - brew cleanup -s
+    - rm -rf $(brew --cache)
+
+install:
+    - |
+      MINICONDA_URL="http://repo.continuum.io/miniconda"
+      MINICONDA_FILE="Miniconda3-latest-MacOSX-x86_64.sh"
+      curl -O "${MINICONDA_URL}/${MINICONDA_FILE}"
+      bash $MINICONDA_FILE -b
+
+      export PATH=/Users/travis/miniconda3/bin:$PATH
+      conda config --add channels conda-forge
+      conda config --set show_channel_urls true
+      conda install --yes --quiet conda-forge-build-setup
+      source run_conda_forge_build_setup
+
+script:
+  - conda build ./recipe
+
+  - ./ci_support/upload_or_check_non_existence.py ./recipe conda-forge --channel=main

--- a/.travis.yml
+++ b/.travis.yml
@@ -37,6 +37,8 @@ install:
       conda config --set show_channel_urls true
       conda install --yes --quiet conda-forge-build-setup
       source run_conda_forge_build_setup
+      # Bypass the control-freak run_conda_forge_build_setup and use versions that do work.
+      conda install --yes conda-build=1.21.11
 
 script:
   - conda build ./recipe

--- a/README.md
+++ b/README.md
@@ -7,7 +7,7 @@ Package license: GPL-2.0
 
 Feedstock license: BSD 3-Clause
 
-Summary: Fortran to Python interface generator with derived type support
+Summary: Fortran to Python interface generator with derived type support.
 
 
 
@@ -71,7 +71,7 @@ Current build status
 ====================
 
 Linux: [![Circle CI](https://circleci.com/gh/conda-forge/f90wrap-feedstock.svg?style=shield)](https://circleci.com/gh/conda-forge/f90wrap-feedstock)
-OSX: ![](https://cdn.rawgit.com/conda-forge/conda-smithy/90845bba35bec53edac7a16638aa4d77217a3713/conda_smithy/static/disabled.svg)
+OSX: [![TravisCI](https://travis-ci.org/conda-forge/f90wrap-feedstock.svg?branch=master)](https://travis-ci.org/conda-forge/f90wrap-feedstock)
 Windows: ![](https://cdn.rawgit.com/conda-forge/conda-smithy/90845bba35bec53edac7a16638aa4d77217a3713/conda_smithy/static/disabled.svg)
 
 Current release info

--- a/ci_support/run_docker_build.sh
+++ b/ci_support/run_docker_build.sh
@@ -41,6 +41,9 @@ conda clean --lock
 conda install --yes --quiet conda-forge-build-setup
 source run_conda_forge_build_setup
 
+# Bypass the control-freak run_conda_forge_build_setup and use versions that do work.
+conda install --yes conda-build=1.21.11
+
 # Embarking on 6 case(s).
     set -x
     export CONDA_NPY=110

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -11,8 +11,9 @@ source:
 
 build:
     number: 0
-    skip: True  # [win or osx]
+    skip: True  # [win]
     script:
+        - export LDFLAGS="-undefined dynamic_lookup -bundle"  # [osx]
         - python setup.py install --single-version-externally-managed --record record.txt
 
 requirements:
@@ -33,11 +34,13 @@ test:
     commands:
         - f90wrap -h
         - f2py-f90wrap -h
+        - conda inspect linkages -n _test f90wrap  # [not win]
+        - conda inspect objects -n _test f90wrap  # [osx]
 
 about:
     home: https://github.com/jameskermode/f90wrap
     license: GPL-2.0
-    summary: Fortran to Python interface generator with derived type support
+    summary: 'Fortran to Python interface generator with derived type support.'
 
 extra:
     recipe-maintainers:

